### ISSUE TITLE
Improve the problem of slow reconstruction loss calculation

### ISSUE THREAD

**Issue by yixinkai123** [XXnormal/RAPIER#9]
def reconstruct(self, x, y):
        # reconstruct层
        torch.cuda.set_device(self.device)
        x_rec = self.rec_fc2(F.selu(self.rec_fc1(x)))
        loss = F.cross_entropy(x_rec.view(-1, 2000), y.view(-1), reduction='none')
        loss = loss.view(-1, 50)
        # loss = torch.stack([
        #     torch.stack([
        #         self.cross_entropy(
        #             x_sft.unsqueeze(0),
        #             y_label.unsqueeze(0)
        #         ) for x_sft, y_label in zip(xi, yi)
        #     ]) for xi, yi in zip(x_rec, y)
        # ], dim=0)
        mask = y.bool()
        loss_ret = torch.sum(loss * mask, dim=1) / torch.sum(mask, dim=1)
        return loss_ret

**Comment by XXnormal** [comment]
感谢修改建议，已修改完成
